proxy-node-shared/src/test/java/uk/gov/ida/notification/saml/EidasResponseBuilderTest.java

```diff
@@ -93,7 +96,13 @@ public void testAllResponseAssertionsAreSet() {
 
         Assertion assertion = assertions.iterator().next();
 
-        assertThat(assertion.getSubject().getNameID().getValue()).contains("an assertion subject");
+        Subject subject = assertion.getSubject();
+        assertThat(subject.getNameID().getValue()).contains("an assertion subject");
+        assertThat(subject.getSubjectConfirmations().size()).isEqualTo(1);
+        SubjectConfirmation subjectConfirmation = subject.getSubjectConfirmations().iterator().next();
+        SubjectConfirmationData subjectConfirmationData = subjectConfirmation.getSubjectConfirmationData();
```

Comment:
I'd do this on one line and use `get(index)` instead of iterator:
`SubjectConfirmationData subjectConfirmationData = subject.getSubjectConfirmations().get(0).getSubjectConfirmationData();`